src/platform/errors/errorHandler.node.ts

```diff
@@ -153,9 +152,10 @@ export class DataScienceErrorHandler implements IDataScienceErrorHandler {
     }
     public async handleKernelError(
         err: Error,
-        errorContext: 'start' | 'restart' | 'interrupt' | 'execution',
+        errorContext: KernelAction,
         kernelConnection: KernelConnectionMetadata,
-        resource: Resource
+        resource: Resource,
+        actionSource: KernelActionSource
```

Comment:
We now need the source of the action, action can be `start, restart, ineterrupt, ` etc.
However depending on who is invoking this action, the outcome could differ (e.g. ability to change kernels if the current kernel doesn't have the necessary dependencies is not an option).